Test/E2ETests/TestApps/NetCore30/E2ETestAppCore30/Controllers/ExternalCallsController.cs

```diff
@@ -0,0 +1,279 @@
+        }
+
+        public ExternalCallsController(IOptions<AppInsightsOptions> options, TelemetryConfiguration telemetryConfiguration)
+        {
+            if (string.IsNullOrEmpty(ConnectionString))
+            {
+                ConnectionString = string.Format(ConnectionStringFormat, options.Value.SqlServerInstance);
+            }
+
+            var webApiHostName = options.Value.Webapihostname;
+            UrlTestWebApiGetCall = string.Format(UrlTestWebApiGetCallTemplate, webApiHostName);
+            UrlWhichReturns500 = string.Format(UrlWhichReturns500Format, webApiHostName);
+            this.telemetryConfiguration = telemetryConfiguration;
+        }
+
+        // GET external/calls
+        [HttpGet]
+        public string Get()
+        {
+            string title = "(No title set)";
+            string response = "(No response set)";
+
+            string type = GetQueryValue("type");
+            string countStr = GetQueryValue("count");
+
+            bool.TryParse(GetQueryValue("success"), out var success);
+            string sqlQueryTouse = success ? ValidSqlQueryToApmDatabase : InvalidSqlQueryToApmDatabase;
+
+            int count;
+            if (!int.TryParse(countStr, out count))
+            {
+                count = 1;
+            }
+
+            switch (type)
+            {
+                case "flush":
+                    title = response = "Flushed telemetry channel";
+                    telemetryConfiguration.TelemetryChannel.Flush();                    
+                    break;
+                case "setsqlserverinstance":
+                    string sqlServerInstance = GetQueryValue("server");
+                    if(!string.IsNullOrEmpty(sqlServerInstance))
+                    {
+                        ConnectionString = string.Format(ConnectionStringFormat, sqlServerInstance);
+                        title = response = "Update SQL Server Instance to: " + sqlServerInstance;
+                    }
+                    else
+                    {
+                        title = response = "SQL Server Instance not updated. ";
+                    }                    
+                    
+                    break;
+                case "setchannelendpoint":
+                    string endPoint = GetQueryValue("endpoint");
+                    if (!string.IsNullOrEmpty(endPoint))
+                    {
+                        telemetryConfiguration.TelemetryChannel.EndpointAddress = string.Format(Program.EndPointAddressFormat, endPoint);
+                        title = response = "Update Endpoint to: " + telemetryConfiguration.TelemetryChannel.EndpointAddress;
+                    }
+                    else
+                    {
+                        title = response = "Endpoint not updated. ";
+                    }
+
+                    break;
+
+                case "http":
+                    title = "Made Sync GET HTTP call to bing";
```

Comment:
Technically this adds a dependency of out tests reliability on Bing's availability, thus lowering ( :) ) our pass rate in the long run. No op, though, our pass rate is less than Bing's availability :)

Reply:
since this is a copy-paste of other func tests, I'll leave it like that. Bing availability is relatively good compared to our tests stability ;)